Sources/DockProgress/DockProgress.swift

```diff
@@ -97,4 +100,46 @@ public final class DockProgress {
 		progressCircle.progress = progressValue
 		progressCircle.render(in: cgContext)
 	}
+	
+	private static func drawProgressBadge(_ dstRect: CGRect, color: NSColor, badgeLabel: Int) {
+		guard let cgContext = NSGraphicsContext.current?.cgContext else {
+			return
+		}
+
+		let radius = dstRect.width / 4.8
+		let newCenter = CGPoint(x: dstRect.maxX - radius - 4, y: dstRect.minY + radius + 4)
+
+		// Background
+		let badge = ProgressCircleShapeLayer(radius: Double(radius), center: newCenter)
+		badge.fillColor = CGColor(red: 0.94, green: 0.96, blue: 1, alpha: 1)
+		badge.shadowColor = .black
+		badge.shadowOpacity = 0.3
+		badge.masksToBounds = false
+		badge.shadowOffset = CGSize(width: -1, height: 1)
+		badge.shadowPath = badge.path
+
+		// Progress circle
+		let lineWidth: CGFloat = 6
+		let innerRadius = Double(radius) - Double(lineWidth / 2)
```

Comment:
It's more efficient to convert to `Double` where it's used (line 124) instead of for each variable.